backend/danswer/datastores/interfaces.py

```diff
@@ -10,5 +11,7 @@ def index(self, chunks: list[EmbeddedIndexChunk]) -> bool:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def search(self, query: str, num_to_retrieve: int) -> list[InferenceChunk]:
+    def semantic_retrieval(
+        self, query: str, filters: dict[str, Any] | None, num_to_retrieve: int
```

Comment:
@Weves is there any built in types option for Any nowadays or do we still need to import this one?

Reply:
afaik, still need to import Any

Reply:
might be better to have an explicit definition of what `filters` is? If I was using a `Datastore` instance, I wouldn't know what I would need to pass into `filters` to do what I want without looking into the implementation of `QdrantDatastore` which seems to defeat the purpose of having the abstraction in the first place

Reply:
Thought process was that filters for other Datastore types could be different, thoughts?